### ISSUE TITLE
Improve eq operator in where function

### ISSUE THREAD

**Issue by jmooring** [gohugoio/hugo#11287]
Use case 1: find all pages where tags equals `['foo']`. 
Use case 2: find all pages where tags equals `['foo','bar']`. 

This is not possible today. You can find all pages where these tags intersect a provided slice, but you cannot compare slices with the `eq` operator.

The `eq` operator _does_ work if your front matter is a scalar, e.g.,

```text
title = 'something 1`
tags = 'foo'
```

But you can't do this:

```text
title = 'something 2`
tags = ['foo']
```

You should be able to get _both_ of the pages above with:

```text
{{ where .Pages "Params.tags" eq "foo" }} OR
{{ where .Pages "Params.tags" eq "foo" (slice "foo"}} 
```

**Comment by bep** [comment]
>Use case 1: find all pages where tags equals ['foo'].
>Use case 2: find all pages where tags equals ['foo','bar'].

The above is rather technical use cases ... 

I can understand if you want to find all pages tagged with `foo`, but finding all pages which has the tags `foo` and `bar` and only those, that's not a use case I have seen.

**Comment by jmooring** [comment]
> finding all pages which has the tags foo and bar and only those, that's not a use case I have seen

Understood. Np problem. If we can do this:

```text
{{ where .Pages "Params.tags" eq "foo" }} OR
{{ where .Pages "Params.tags" eq "foo" (slice "foo"}} 
```

...then we can do it again again with "bar" and `union` the results.

**Comment by jmooring** [comment]
Go 1.21 introduced the slices package which contains https://pkg.go.dev/slices#Equal.

**Comment by bep** [comment]
>Go 1.21 introduced the slices package which contains https://pkg.go.dev/slices#Equal.

The functional part of this has never been the challenge. If you look at that implementation you will find a ... loop.

**Comment by jmooring** [comment]
Understood. Will stick with...

```text
{{ $p := where site.RegularPages "Params.tags" "intersect" (slice "a" "b") }}
{{ range $p }}
  {{ if eq (len .Params.tags) 2 }}
    <h2><a href="{{ .RelPermalink }}">{{ .LinkTitle }}</a></h2>
  {{ end }}
{{ end }}
```

**Comment by bep** [comment]
To be clear:

>tags = ['foo', 'bar']

>{{ where .Pages "Params.tags" eq (slice "foo" "bar" }} 

Would be cheapish to implement with strict ordering (['foo', 'bar'] != ['bar', 'foo']), but I'm not sure how useful that would be.

**Comment by jmooring** [comment]
I agree that strict ordering limits its usage; not sure I'd pursue it.
